### ISSUE TITLE
Upgrading to babel 5.5.1 breaks configuration

### ISSUE THREAD

**Issue by Dignifiedquire** [babel/babel-loader#76]
I've been using `babel-loader@5.1.3` with `babel@5.4.7` without any issues, but upgrading to babel breaks all my configuration in `.babelrc`.
Specifically: 
* Previously ignored files are not ignored anymore
* When running the code it throws an error `require is not defined` (which I expect is a sign that the runtime is not injected properly?)

Below the configuration

```js
// .babelrc
{
  "stage": 0,
  "optional": [
    "runtime"
  ],
  "ignore": [
    "scripts/vendors/primus.js",
    "scripts/vendors/gunzip.js",
    "node_modules/worker-loader/index.js?inline!*"
  ]
}
```
```js
// webpack
{
  test: /react\-infinite[\/\\]src[\/\\].*\.js/,
  loader: 'babel-loader'
}, {
  test: /\.js$/,
  exclude: /node_modules/,
  loader: 'babel-loader'
}, {
  test: /worker\.js$/,
  exclude: /node_modules/,
  loader: 'worker?inline!babel-loader'
}
```

**Comment by Couto** [comment]
Does it work fine if you use the babel cli?

**Comment by Dignifiedquire** [comment]
I does run, but I can't test it fully with the cli as the code depends on more webpack specific loaders.

**Comment by Dignifiedquire** [comment]
Using the `require` hook from babel in my tests, seems to work though.

**Comment by Couto** [comment]
Ok! Thanks for the tests.
I'll try to replicate your problem as soon as I can (which means when I get out of work).

Also out of curiosity, and kinda like a shot in the dark, can you try the new loader version? (5.1.4)
It fixes an unrelated issue, but the timeline of both issues is the same, so I'm just throwing ideas until I get to fix your issue.

**Comment by Dignifiedquire** [comment]
Thanks for the quick reply. I did some more debugging and this is what I found:

* 5.1.4 no changes in behaviour.
* Logging `opts` in https://github.com/babel/babel/blob/master/src/babel/transformation/transformer-pipeline.js#L70 does not show the `.babelrc` options (they are not presented in any of the babel-loader code either, but I expected that as `babel` should load them?)

**Comment by Dignifiedquire** [comment]
* Hard coding the options in the webpack configuration makes them show up in the above mentioned place, but they still seem to not have any effect :(

**Comment by sebmck** [comment]
Fixed as of Babel 5.5.4, thanks! :)
